apps/frontend/src/pages/management/libs/components/groups-tab/groups-tab.tsx

```diff
@@ -0,0 +1,261 @@
+import { EditModal } from "../edit-modal/edit-modal.js";
+import { Table, TableCell, TableRow } from "../table/table.js";
+import { GroupsTableHeader } from "./libs/enums/enums.js";
+import { groupsHeaderToPropertyName } from "./libs/maps/maps.js";
+import styles from "./styles.module.css";
+
+const GroupsTab: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const { authUser, groups, permissions } = useAppSelector((state) => {
+		return {
+			authUser: state.auth.user as UserAuthResponseDto,
+			groups: state.management.groups,
+			permissions: state.management.permissions,
+		};
+	});
+	const { control, errors, handleSubmit, reset } = useAppForm<GroupRequestDto>({
+		defaultValues: {
+			key: "",
+			name: "",
+		},
+		validationSchema: groupNameFieldValidationSchema,
+	});
+
+	const [currentGroup, setCurrentGroup] = useState<GroupResponseDto | null>(
+		null,
+	);
+	const [isEditModalOpen, setEditModalOpen] = useState<boolean>(false);
+	const [isConfirmationModalOpen, setConfirmationModalOpen] =
+		useState<boolean>(false);
+
+	useEffect(() => {
+		void dispatch(groupsActions.getAllGroups());
+		void dispatch(permissionsActions.getAllPermissions());
+	}, [dispatch]);
+
+	const handleCreateGroup = useCallback(
+		(group: GroupRequestDto) => {
+			const key = group.name.trim().replace(" ", "-").toLowerCase();
+			void dispatch(groupsActions.createGroup({ key, name: group.name }));
+			reset();
+		},
+		[dispatch, reset],
+	);
+
+	const handleFormSubmit = useCallback(
+		(event_: React.BaseSyntheticEvent): void => {
+			void handleSubmit(handleCreateGroup)(event_);
+		},
+		[handleSubmit, handleCreateGroup],
+	);
+
+	const handleOpenEditModal = useCallback((group: GroupResponseDto) => {
+		return () => {
+			setCurrentGroup(group);
+			setEditModalOpen(true);
+		};
+	}, []);
+
+	const handleCloseEditModal = useCallback(() => {
+		setEditModalOpen(false);
+		setCurrentGroup(null);
+	}, []);
+
+	const handleOpenConfirmationModal = useCallback((group: GroupResponseDto) => {
+		return () => {
+			setCurrentGroup(group);
+			setConfirmationModalOpen(true);
+		};
+	}, []);
+
+	const handleCloseConfirmationModal = useCallback(() => {
+		setConfirmationModalOpen(false);
+		setCurrentGroup(null);
+	}, []);
+
+	const handleChangeGroupPermissions = useCallback(
+		(groupId: number, permissionId: number) => {
+			void dispatch(
+				groupsActions.updateGroupPermissions({ groupId, permissionId }),
+			);
+		},
+		[dispatch],
+	);
+
+	const handleToggleCheckbox = useCallback(
+		(permissionId: number) => {
+			handleChangeGroupPermissions(currentGroup?.id as number, permissionId);
+		},
+		[currentGroup, handleChangeGroupPermissions],
+	);
+
+	const handleDeleteGroup = useCallback(
+		(groupId: number) => {
+			return () => {
+				void dispatch(groupsActions.deleteGroup(groupId));
+				handleCloseConfirmationModal();
+			};
+		},
+		[dispatch, handleCloseConfirmationModal],
+	);
+
+	const tableHeaders = [
```

Comment:
```suggestion
	const tableHeaders = getTableHeaders()
```
move to component's helpers

Reply:
Can we use this array as a constant in libs here? 
I think this variable doesn't contain any reusable logic, it's used only for the right headers order.

Reply:
can be a constant